source/Stargate-Metrics/Counter.class.st

```diff
@@ -0,0 +1,70 @@
+"
+A counter is a cumulative metric that represents a single monotonically increasing counter whose value can only increase or be reset to zero on restart. 
+
+For example, you can use a counter to represent the number of requests served, tasks completed, or errors.
+Do not use a counter to expose a value that can decrease. For example, do not use a counter for the number of currently running processes; instead use a gauge.
+"
+Class {
+	#name : #Counter,
+	#superclass : #Metric,
+	#instVars : [
+		'name',
+		'description',
+		'value'
+	],
+	#category : #'Stargate-Metrics'
+}
+
+{ #category : #'instance creation' }
+Counter class >> named: aMetricName describedBy: aMetricDescription withValue: aNumber [
+
+	AssertionCheckerBuilder new
+		raising: InstanceCreationFailed;
+		checking: [ :asserter | 
+			asserter
+				enforce: [ aMetricName notEmpty ] because: 'A metric name cannot be empty';
```

Comment:
This is not "a metric". We know which one we are trying to create.
If the idea is to have some general reusable rules, the assertion should be extracted somewhere else.